core/src/main/scala/org/http4s/otel4s/middleware/ClientMiddleware.scala

```diff
@@ -151,18 +158,33 @@ object ClientMiddleware {
             traceHeaders <- Resource.eval(Tracer[F].propagate(Headers.empty)).mapK(trace)
             newReq = req.withHeaders(traceHeaders ++ req.headers)
 
-            _ <- Resource.onFinalizeCase { ec =>
-              span.addAttribute(CustomAttributes.exitCase(ec.toOutcome))
+            resp <- poll(client.run(newReq).mapK(trace)).guaranteeCase {
+              case Outcome.Succeeded(fa) =>
+                fa.evalMap { resp =>
+                  val out = response(resp, allowedResponseHeaders) ++
+                    additionalResponseAttributes(resp.responsePrelude)
+
+                  span.addAttributes(out) >> span
+                    .setStatus(StatusCode.Error)
+                    .unlessA(resp.status.isSuccess)
```

Comment:
We should set the status to `Error` if error code is 4xx-5xx - https://opentelemetry.io/docs/specs/semconv/http/http-spans/#http-client:

> If the request fails with an error before response status code was sent or received, error.type SHOULD be set to exception type (its fully-qualified class name, if applicable) or a component-specific low cardinality error identifier.
> 
> If response status code was sent or received and status indicates an error according to [HTTP span status definition](https://opentelemetry.io/docs/specs/semconv/http/http-spans/), error.type SHOULD be set to the status code number (represented as a string), an exception type (if thrown) or a component-specific error identifier.
> 
> The error.type value SHOULD be predictable and SHOULD have low cardinality. Instrumentations SHOULD document the list of errors they report.
> 
> The cardinality of error.type within one instrumentation library SHOULD be low, but telemetry consumers that aggregate data from multiple instrumentation libraries and applications should be prepared for error.type to have high cardinality at query time, when no additional filters are applied.
> 
> If the request has completed successfully, instrumentations SHOULD NOT set error.type.